File: C#/Banco/TarjetaDeDebito.cs
using System;
using System.Collections.Generic;
using System.Text;

using Banco.State;

namespace Banco
{
    public class TarjetaDeDebito
    {

        private string nombre;
        private string numero;
        private IEstado estado = new Normal();

        public TarjetaDeDebito(string nombre, string numero)
        {
            this.nombre = nombre;
            this.numero = numero;
        }

        public void retirarSaldo(float saldoARetirar)
        {
            estado.retirarSaldo(saldoARetirar);
        }

        public void cambiarEstado(IEstado estado)
        {
            this.estado = estado;
        }

    }
}

File: Java/src/main/java/banco/TarjetaDeDebito.java
package banco;

import banco.State.IEstado;
import banco.State.Normal;

public class TarjetaDeDebito {
    
    private String nombre;
    private String numero;
    private IEstado estado = new Normal();

    public TarjetaDeDebito(String nombre, String numero)
    {
        this.nombre = nombre;
        this.numero = numero;
    }

    public void retirarSaldo(float saldoARetirar)
    {
        this.estado.retirarSaldo(saldoARetirar);
    }

    public void cambiarEstado(IEstado estado)
    {
        this.estado = estado;
    }

}